pages/profile/[username].tsx

```diff
@@ -0,0 +1,70 @@
+import { TableContainer } from "@chakra-ui/table";
+import { DataTable } from "components/shared/DataTable";
+import { PageContainer } from "components/shared/PageContainer";
+import { Views } from "queries";
+import { getSupabaseClient } from "utils/getSupabaseClient";
+
+const dbClient = getSupabaseClient();
+
+export async function getStaticPaths() {
+  const { data } = await dbClient.from(Views.PublicSignups).select("username");
+
+  const payload = {
+    paths: data?.map(({ username }) => ({
+      params: { username },
+    })),
+    fallback: false, // can also be true or 'blocking'
+  };
+  return payload;
+}
+
+// `getStaticPaths` requires using `getStaticProps`
+export async function getStaticProps({
+  params: { username },
+}: {
+  params: { username: string };
+}) {
+  const { data } = await dbClient
+    .from(Views.PublicSubmissions)
+    .select("*")
+    .filter("username", "eq", username);
+
+  return {
+    // Passed to the page component as props
+    props: {
+      username,
+      data,
+    },
+  };
+}
+
+export default function Profile(props: { username: string; data: any }) {
```

Comment:
So did you rip this function from the profiles component? I was i got really confused if we had to get rid of the router in that component.

Reply:
Lol actually yeah - the component name here doesn't matter at all though. All we need is something default exported.